File: misc/js/leap.js
function solutionOne(year) {
  if (year % 400 == 0) {
    console.log("yes! " + year + " is a leap year.");
  } else if (year % 4 == 0) {
    if (year % 100 != 0) {
      console.log("yes! " + year + " is a leap year.");
    } else {
      console.log("nope " + year + " is not a leap year.");
    }
  } else {
    console.log("nope " + year + " is not a leap year.");
  }
}

function solutionTwo(year) {
  if (year % 400 == 0 || (year % 4 == 0) && (year % 100 != 0)) {
    console.log("yes! " + year + " is a leap year.");
  } else {
    console.log("nope " + year + " is not a leap year.");
  }
}

input = process.argv[2];
solutionOne(input);
solutionTwo(input);

File: misc/ruby/leap.rb
def solution_one(year)
  year = year.to_i
  if year % 400 == 0
    puts "yes! #{year} is a leap year."
  elsif year % 4 == 0
    if year % 100 != 0
      puts "yes! #{year} is a leap year."
    else
      puts "nope, #{year} is not a leap year."
    end
  else
    puts "nope, #{year} is not a leap year."
  end
end

def solution_two(year)
  year = year.to_i
  if year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)
    puts "yes! #{year} is a leap year!"
  else
    puts "nope, #{year} is not a leap year."
  end
end

input = ARGV[0]
solution_one(input)
solution_two(input)
